06-building-a-program-start-to-finish/solving-the-problem/Output/invertedIndex.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Inverted index class
+ */
+class InvertedIndex {
+  /**
+   * Inverted index constructor
+   */
+  constructor() {
+    // Object to hold the index
+    this.index = {};
+  }
+
+  /**
+   * @param{String} words - String to tokenize
+   * @return{Array} list of words devoid of special characters or symbols
+   */
+  static tokenize(words) {
+    return words.trim().replace(/-/g, ' ')
+      .replace(/[^A-z\s]/g, '')
+      .toLowerCase()
+      .split(' ')
+      .sort();
+  }
+
+  /**
+   * @param{String} words - The string to be filtered
+   * @return{Array} tokens - Without duplicated words
+   */
+  static uniqueWords(words) {
+    const tokens = InvertedIndex.tokenize(words);
+    return tokens.filter((item, index) => tokens.indexOf(item) === index);
+  }
+
+  /**
+   * @param{String} fileName - The name of the file to be indexed
+   * @param{Array} fileToIndex - Array of contents of the JSON file to index
+   * @return{Object} index - That maps words to locations(documents)
+   */
+  createIndex(fileName, fileToIndex) {
+    const wordsToIndex = [];
+    const fileIndex = {};
+    const fileLength = fileToIndex.length;
+    if (fileLength === 0) {
+      return 'JSON file is Empty';
```

Comment:
Your JSDoc says you expect this function to return an Object `index`. You're returning a string here when the file is invalid, is there a way to represent this?